Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs

```diff
@@ -192,13 +199,33 @@ private IEnumerator TryToReconnect()
                 NetworkManager.Singleton.Shutdown();
                 yield return new WaitWhile(() => NetworkManager.Singleton.ShutdownInProgress); // wait until NetworkManager completes shutting down
                 Debug.Log($"Reconnecting attempt {nbTries + 1}/{k_NbReconnectAttempts}...");
-                ConnectClient(null);
+                if (!string.IsNullOrEmpty(lobbyCode))
+                {
+                    var leavingLobby = true;
+                    m_LobbyServiceFacade.ForceLeaveLobbyAttempt(() => leavingLobby = false, () => leavingLobby = false);
+                    yield return new WaitWhile(() => leavingLobby); // This is not a clean way of doing this, ideally we would want to replace that logic with a proper await.
+                    // todo should use yield return new WaitUntil(() => task.IsCompleted);
+                    var joiningLobby = true;
+                    m_LobbyServiceFacade.JoinLobbyAsync("", lobbyCode, onSuccess: lobby =>
+                        {
+                            m_LobbyServiceFacade.BeginTracking(lobby);
+                            ConnectClient(null);
+                            joiningLobby = false;
+                        }
+                        , onFailure: () => joiningLobby = false);
+                    yield return new WaitWhile(() => joiningLobby); // This is not a clean way of doing this, ideally we would want to replace that logic with a proper await.
```

Comment:
If you need the task object - then I suggest modifying the `LobbyAPIInterface` in such a way so that instead of returning `void` all over the place it would return the Task object that it creates. This way you'd be able to avoid this whole pattern with `joiningLobby = true` and the two functions that switch it to false.

Reply:
I was thinking of leaving it this way for the moment since #501 already aims to do that change and I didn't want to step on any toes, but anyways this specific part won't be going to develop immediately since auto-reconnect won't be there until pre.7 at least. So in that case it shouldn't create conflicts